### DISCUSSION TITLE
Is there anyway to run actions across organisations

### DISCUSSION THREAD

**Discussion by andypandy-tappit** [zitadel/zitadel#8771] [Q&A] [answered]
Hi there,

We've setup multiple Zitadel organisations to enable multi-tenancy within our system. On the root org, we've added in an action which enriches user tokens with various claims that are then parsed and used in our backend services to help prevent chatty services. Unfortunately, any users created within a tenant org do not have their tokens enriched despite logging into an application held within the root organisation that contains this action.

I can't see anyway to enable to tenant users to benefit from the single action in the root org without duplicating the action into their organisation. This is problematic for us as if any of the logic were to change, we'd then have to update every single action across all orgs. Considering the user is granted access to and is logging into the root org application, is there anyway to run the root org actions on that user logging in?

We are currently exploring ways around this  (using user metadata or enriching the token elsewhere) but the ideal scenario in my mind is to have a single action run across many orgs.

Thanks

**Comment by hifabienne** [comment, marked as answer]
Hi @andypandy-tappit 
At the moment there is no other option than adding the action on each organization.
With the new version of Actions (Action V2), we will enable Actions on an instance level. 
Wich means you can add it once and it is for all organizations.
But this is not done yet

**Reply by andypandy-tappit** [reply]
Thanks for your response, that makes sense. Looking forward to the V3 api being generally available!

**Comment by sriattypelink** [comment]
Any ETA on this by any chance?

**Reply by hifabienne** [reply]
part of the v3 actions is already implemented, the rest should be available by end of q1 - 25.

**Comment by firstdorsal** [comment]
Hi, will it be possible with the new Action V2 API to create different actions for different projects in the same org?

**Reply by hifabienne** [reply]
The actions v2 will run on the instance, however you are able to implement what ever you need in there.
So you can implement in the action, that it should only run when the project is a certain id.
